fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/patch/Main.java

```diff
@@ -0,0 +1,336 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.schema.patch;
+
+import static com.ibm.fhir.schema.app.util.CommonUtil.configureLogger;
+import static com.ibm.fhir.schema.app.util.CommonUtil.getDbAdapter;
+import static com.ibm.fhir.schema.app.util.CommonUtil.getPropertyAdapter;
+import static com.ibm.fhir.schema.app.util.CommonUtil.loadDriver;
+import static com.ibm.fhir.schema.app.util.CommonUtil.logClasspath;
+import static com.ibm.fhir.schema.app.util.CommonUtil.printUsage;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.ibm.fhir.database.utils.api.DatabaseNotReadyException;
+import com.ibm.fhir.database.utils.api.IDatabaseAdapter;
+import com.ibm.fhir.database.utils.api.IDatabaseTranslator;
+import com.ibm.fhir.database.utils.common.DataDefinitionUtil;
+import com.ibm.fhir.database.utils.common.DropForeignKeyConstraint;
+import com.ibm.fhir.database.utils.common.JdbcPropertyAdapter;
+import com.ibm.fhir.database.utils.common.JdbcTarget;
+import com.ibm.fhir.database.utils.db2.Db2Translator;
+import com.ibm.fhir.database.utils.derby.DerbyTranslator;
+import com.ibm.fhir.database.utils.model.DbType;
+import com.ibm.fhir.database.utils.postgresql.PostgreSqlTranslator;
+import com.ibm.fhir.model.type.code.FHIRResourceType;
+
+/**
+ * Utility to patch an old schema for use-cases which fall outside the standard
+ * automatic schema migration process
+ */
+public class Main {
```

Comment:
is this a "3rd way" to do schema migrations?  i was already concerned about supporting two different ways...

Reply:
Same concerns here.

Reply:
No choice. One of the dev databases had cruft left over from a previous design that our current migration code doesn't know how to handle. It's a one-off, and not part of the standard mechanism.